bug: Non-Deterministic Test RoutingPassThroughIT.shouldVirtualizeNodeIdsInDescribeTopicPartitionsResponse

A test failed non-deterministically in GitHub Actions:

## failed test

RoutingPassThroughIT.shouldVirtualizeNodeIdsInDescribeTopicPartitionsResponse
https://github.com/kroxylicious/kroxylicious/actions/runs/29322907915/job/87054012389?pr=4361

## stack trace

```
java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.UnknownTopicOrPartitionException: This server does not host this topic-partition.
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at org.apache.kafka.common.internals.KafkaFutureImpl.get(KafkaFutureImpl.java:173)
	at io.kroxylicious.it.RoutingPassThroughIT.shouldVirtualizeNodeIdsInDescribeTopicPartitionsResponse(RoutingPassThroughIT.java:422)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: org.apache.kafka.common.errors.UnknownTopicOrPartitionException: This server does not host this topic-partition.
```

The test creates a topic and then issues a `DescribeTopicPartitions` request, but the broker reports `UnknownTopicOrPartitionException`, suggesting the topic metadata has not fully propagated across the cluster at the point the describe is attempted.

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/RoutingPassThroughIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/RoutingPassThroughIT.java
@@ -63,6 +63,7 @@ import static io.kroxylicious.testing.integration.tester.KroxyliciousConfigUtils
 import static io.kroxylicious.testing.integration.tester.KroxyliciousConfigUtils.defaultGatewayBuilder;
 import static io.kroxylicious.testing.integration.tester.KroxyliciousConfigUtils.defaultPortIdentifiesNodeGatewayBuilder;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
 
 /**
  * Integration tests verifying that routing via a pass-through router
@@ -417,14 +418,16 @@ class RoutingPassThroughIT {
 
         try (var tester = KroxyliciousTesters.newBuilder(config).setFeatures(ROUTING_ENABLED).createDefaultKroxyliciousTester();
                 var admin = tester.admin()) {
+            // Given
             admin.createTopics(List.of(new NewTopic(topicName, 1, (short) 1))).all().get(10, TimeUnit.SECONDS);
-            // When
-            var descriptions = admin.describeTopics(List.of(topicName)).allTopicNames().get(10, TimeUnit.SECONDS);
 
-            // Then: leader ID is the virtual ID (1), not the upstream broker ID (0)
-            assertThat(descriptions.get(topicName).partitions())
-                    .singleElement()
-                    .satisfies(p -> assertThat(p.leader().id()).isEqualTo(1));
+            // When / Then
+            await().atMost(Duration.ofSeconds(30)).untilAsserted(() -> {
+                var descriptions = admin.describeTopics(List.of(topicName)).allTopicNames().get(10, TimeUnit.SECONDS);
+                assertThat(descriptions.get(topicName).partitions())
+                        .singleElement()
+                        .satisfies(p -> assertThat(p.leader().id()).isEqualTo(1));
+            });
         }
     }
 

